upi/openstack/network.yaml

```diff
@@ -114,6 +126,40 @@
       cmd: "openstack router set --tag {{ cluster_id_tag }} {{ os_router }}"
     when: os_external_network is defined and os_external_network|length>0
 
+  - name: 'Get information about router'
+    ansible.builtin.command:
+      cmd: "openstack router list --name {{ subnet_pool }} --tags {{ cluster_id_tag }}"
+    register: router
+    when:
+    - os_subnet6_range is defined
+    - "{{ os_subnet6_range|ansible.utils.ipv6 }}"
+
+  - name: 'Create external router IPv6'
+    openstack.cloud.router:
+      name: "{{ os_router }}"
+      network: "{{ os_network }}"
+      interfaces:
+      - "{{ os_subnet6 }}"
+    when:
+    - os_subnet6_range is defined
+    - "{{ os_subnet6_range|ansible.utils.ipv6 }}"
+    - router.stdout == ""
+
+  - name: 'Set external router IPv6 tag'
+    ansible.builtin.command:
+      cmd: "openstack router set --tag {{ cluster_id_tag }} {{ os_router }}"
+    when:
+    - os_subnet6_range is defined
+    - "{{ os_subnet6_range|ansible.utils.ipv6 }}"
+    - router.stdout == ""
+
+  - name: 'Set router external gateways'
```

Comment:
It's not possible to set an external-gateway with a network that is not external. So, there is no need to add the IPv6 here given we use an internal network

Reply:
Ack